guard/src/commands/migrate_tests.rs

```diff
@@ -141,4 +176,48 @@ rule aws_s3_bucket_checks WHEN %aws_s3_bucket NOT EMPTY {
     assert_eq!(result, expected_rule);
     rules_file(span)?;
     Ok(())
-}
\ No newline at end of file
+}
+
+#[test]
+fn test_migrate_basic_rules_with_custom_messages() -> Result<()> {
+    let old_ruleset = String::from(
+        r#"AWS::S3::Bucket Foo.Bar == 2 << this must equal 2"#
+    );
+    let rule_lines = parse_rules_file(&old_ruleset, &String::from("test-file")).unwrap();
+    let result = migrate_rules(rule_lines).unwrap();
+    let span = crate::rules::parser::Span::new_extra(&result, "");
+
+    let expected_rule = String::from("let aws_s3_bucket = Resources.*[ Type == \"AWS::S3::Bucket\" ]
+rule aws_s3_bucket_checks WHEN %aws_s3_bucket NOT EMPTY {
+    %aws_s3_bucket {
+        Properties.Foo.Bar == 2 <<this must equal 2>>
+    }
+}
+\n");
+    assert_eq!(result, expected_rule);
+    rules_file(span)?;
+    Ok(())
+}
+
+#[test]
+fn test_migrate_disjunction_basic_and_conditional() -> Result<()> {
+    let old_ruleset = String::from(
+        r#"AWS::EC2::Instance WHEN InstanceType == "m2.large" CHECK .DeletionPolicy == Retain |OR| AWS::EC2::Instance InstanceType == "t2.micro" << Deletion policy for m2.large"#
+    );
+    let rule_lines = parse_rules_file(&old_ruleset, &String::from("test-file")).unwrap();
+    let result = migrate_rules(rule_lines).unwrap();
+    let span = crate::rules::parser::Span::new_extra(&result, "");
+
+    let expected_rule = String::from("let aws_ec2_instance = Resources.*[ Type == \"AWS::EC2::Instance\" ]
+rule aws_ec2_instance_checks WHEN %aws_ec2_instance NOT EMPTY {
+    %aws_ec2_instance {
+        when Properties.InstanceType == \"m2.large\" {
+            DeletionPolicy == \"Retain\"
+        } or Properties.InstanceType == \"t2.micro\" <<Deletion policy for m2.large>>
+    }
+}
+\n");
+    assert_eq!(result, expected_rule);
```

Comment:
Comparing strings makes it very difficult to maintain. Create a `RulesFile` object from both migrated and expected strings and compares these objects